src/test/java/io/supertokens/test/jwt/storage/JWTKeyStorageTest.java

```diff
@@ -0,0 +1,271 @@
+
+                                    if (!success) {
+                                        fail();
+                                    }
+
+                                    endValueOfCon1.set(keyId + "-1");
+                                } else {
+                                    endValueOfCon1.set(matchingKeyInfo.keyId);
+                                }
+
+                                break;
+                            } catch (DuplicateKeyIdException e) {
+                                continue;
+                            }
+                        }
+                    } catch (Exception ignored) {
+
+                    }
+                }
+            };
+
+            Runnable runnable2 = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        while (true) {
+                            numberOfIterations.getAndIncrement();
+                            List<JWTSigningKeyInfo> keysFromStorage = noSQLStorage_1.getJWTSigningKeys_Transaction();
+
+                            JWTSigningKeyInfo matchingKeyInfo = null;
+
+                            for (int j = 0; j < keysFromStorage.size(); j ++) {
+                                if (keysFromStorage.get(j).algorithm.equals(algorithm)) {
+                                    matchingKeyInfo = keysFromStorage.get(j);
+                                    break;
+                                }
+                            }
+
+                            if (numberOfIterations.get() <= 2) {
+                                assert matchingKeyInfo == null;
+                            } else {
+                                assert matchingKeyInfo != null;
+                            }
+
+                            try {
+                                Thread.sleep(700);
+                            } catch (InterruptedException e) {
+                                // Ignore
+                            }
+
+                            try {
+                                if (matchingKeyInfo == null) {
+                                    boolean success = noSQLStorage_1.setJWTSigningKeyInfoIfNoKeyForAlgorithmExists_Transaction(new JWTSymmetricSigningKeyInfo(keyId + "-2", 1000, algorithm, "SomeKeyString"));
+
+                                    if (!success) {
+                                        continue;
+                                    }
+
+                                    endValueOfCon2.set(keyId + "-2");
+                                } else {
+                                    endValueOfCon2.set(matchingKeyInfo.keyId);
+                                }
+
+                                break;
+                            } catch (DuplicateKeyIdException e) {
+                                continue;
+                            }
+                        }
+                    } catch (Exception ignored) {
+
+                    }
```

Comment:
Why do we need this try / catch?

Reply:
This is because the calls to storage can throw exceptions (StorageQueryException for example), the test has been refactored to result in failure if any unexpected exceptions occur